tests/manage/test_pvc_disruptive.py

```diff
@@ -21,43 +61,171 @@ class BaseDisruption(ManageTest):
+        pod_objs = []
+        for pvc_obj in pvc_objs_list:
+            pod_obj = helpers.create_pod(
+                interface_type=constants.CEPHBLOCKPOOL, pvc_name=pvc_obj.name, wait=False, namespace=self.namespace
+            )
+            pod_objs.append(pod_obj)
+        return pod_objs
+
     def disruptive_base(self, operation_to_disrupt, resource_to_delete):
         """
         Base function for PVC disruptive tests
+        Deletion of 'resource_to_delete' will be introduced while 'operation_to_disrupt' is progressing
         """
+        # Fetch the number of Pods and PVCs
+        initial_num_of_pods = len(pod.get_all_pods(namespace=self.namespace))
+        initial_num_of_pvc = len(get_all_pvcs(namespace=self.namespace)['items'])
+
+        executor = ThreadPoolExecutor(max_workers=1)
+
         DISRUPTION_OPS.set_resource(resource=resource_to_delete)
-        self.pvc_obj = helpers.create_pvc(
-            sc_name=self.sc_obj.name, wait=False
-        )
+
+        # Start creation of multiple PVCs
+        bulk_pvc_create = executor.submit(self.create_pvcs)
+
         if operation_to_disrupt == 'create_pvc':
-            DISRUPTION_OPS.delete_resource()
-        self.pvc_obj.reload()
-        assert self.pvc_obj.ocp.wait_for_resource(
-            condition=constants.STATUS_BOUND, resource_name=self.pvc_obj.name, timeout=120
-        )
-
-        self.pod_obj = helpers.create_pod(
-            interface_type=constants.CEPHBLOCKPOOL, pvc_name=self.pvc_obj.name, wait=False
-        )
+            # Ensure PVCs are being created before deleting the resource
+            for count in range(10):
+                current_num_of_pvc = len(get_all_pvcs(namespace=self.namespace)['items'])
+                if current_num_of_pvc > initial_num_of_pvc:
+                    num_of_new_pvc = current_num_of_pvc - initial_num_of_pvc
+                    logging.info(f"PVCs creation has started. Found {num_of_new_pvc} new PVC.")
+                    DISRUPTION_OPS.delete_resource()
+                    break
+                assert count < 9, "PVCs are not being created."
+                logging.info("Creation of PVCs has not started. Checking again.")
+
+        pvc_objs = bulk_pvc_create.result()
+
+        # Verify PVCs are Bound
+        for pvc_obj in pvc_objs:
+            pvc_obj.reload()
+            assert pvc_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_BOUND, resource_name=pvc_obj.name, timeout=120
+            )
+        logging.info("Verified: PVCs are Bound.")
+
+        # Start creating pods
+        bulk_pod_create = executor.submit(self.create_pods, pvc_objs)
+
         if operation_to_disrupt == 'create_pod':
-            DISRUPTION_OPS.delete_resource()
-        self.pod_obj.reload()
+            for count in range(10):
+                # Ensure that pods are being created before deleting the resource
+                current_num_of_pod = len(pod.get_all_pods(namespace=self.namespace))
```

Comment:
The same as my suggestion above, use TimeoutSampler

Reply:
Or even better, create helper method in this class with the TimeoutSampler implementation, that will get the resource kind (pod/pvc) and call it from both places

Reply:
Done. Using TimeoutSampler now.